### DISCUSSION TITLE
What is the branch pattern syntax?

### DISCUSSION THREAD

**Discussion by jcornaz** [semantic-release/semantic-release#2439] [Q&A] [answered]
On the [documentation](https://semantic-release.gitbook.io/semantic-release/usage/configuration#branches) I see that the default branch config uses some kind of pattern such as `+([0-9])?(.{+([0-9]),x}).x`.

But as that is not a valid regex, I am wondering: what is the branch pattern syntax?

**Comment by jcornaz** [comment, marked as answer]
I finally found it. It is a [micromatch glob](https://github.com/micromatch/micromatch#matching-features)

**Reply by 20BBrown14** [reply]
THanks for this, I was wondering myself. I think a small note should be made in the docs for this as it was very confusing at first.

**Reply by travi** [reply]
We would welcome a PR suggesting a clarification like this

**Reply by kilianpaquier** [reply]
Doesn't seem to have been made to this day. Shall we add it everywhere this glob is mentioned ?

**Reply by almcaffee** [reply]
How is this implemented? @jcornaz I don't see details in the link

**Reply by jcornaz** [reply]
@almcaffee I don't know more than you. But the link points to a github repo, so you can go read the code.
